app/com/github/pedrovgs/dotto/core/DottoApp.scala

```diff
@@ -0,0 +1,26 @@
+package com.github.pedrovgs.dotto.core
+
+import cats.free.Free
+import com.github.pedrovgs.dotto.core.algebra.dotto._
+
+/**
+  * Declares the two programs Dotto implements.
+  */
+object DottoApp {
+
+  def enqueueMessageProgram: Message => Free[Instruction, Message] = {
```

Comment:
Seems like you have a single algebra you may just type alias your Free stuff to have a single arg type constructor which is always easy to deal with:
```scala
type InstructionIO[A] = Free[Instruction, A]
```
You can use that alias now through the App since it's assumed `Instruction` is the Algebra you are using anyway.